### ISSUE TITLE
[no-implied-eval] False positive when using setTimeout from timers/promises

### ISSUE THREAD

**Issue by almeidx** [typescript-eslint/typescript-eslint#3727]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-implied-eval": 2
  }
}
```

```TS
import { setTimeout } from 'timers/promises';
// or 
import { setTimeout } from 'node:timers/promises';

// Errors with: Implied eval. Consider passing a function.
setTimeout(1000);
```

**Expected Result**

I expected there to be no errors.

**Actual Result**

The no-implied-eval rule falsely triggers on the setTimeout function

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.29.1` |
| `@typescript-eslint/parser`        | `4.29.1` |
| `TypeScript`                       | `4.3.5` |
| `ESLint`                           | `7.32.0` |
| `node`                             | `16.6.1` |

**Comment by bradzacher** [comment]
The rule currently does not support functions that shadow the global functions and change their functionality.
